package/run.ps1

```diff
@@ -93,11 +93,14 @@ if ($newEnv -and ($newHash -ne $currentHash)) {
 Write-LogInfo "Checking if RKE2 agent service exists"
 if ((Get-Service -Name $rke2ServiceName -ErrorAction SilentlyContinue)) {
     Write-LogInfo "RKE2 agent service found, stopping now"
+    # allow some time for the service to come up, so we can  then properly stop it
+    Start-Sleep -s 5
     Stop-Service -Name $rke2ServiceName
     while ((Get-Service $rke2ServiceName).Status -ne 'Stopped') {
         Write-LogInfo "Waiting for RKE2 agent service to stop"
-        Start-Sleep -s 5
     }
+    # allow time for all processes to stop, and for ports to be freed
+    Start-Sleep -s 30
```

Comment:
This is fine as an interim solution but I would love to be able to see a way to validate that the processes have stopped and the ports have freed -- on a super highly loaded system I worry that 30 seconds may not even be enough, I do know below you are retrying startup so there is that, but this is artificially slowing things down potentially.